### PR TITLE
Confirm empty question sets collection established at Person init

### PR DESCRIPTION
**Pull request by tjchambers**
This spec change confirms person initialization also creates a question set collection array. This kills an alive mutant, and besides one other creation sequencing issue ensures Person class is mutation covered.

### CODE REVIEW

**Comment by tjchambers** [comment]
@dkubb assuming green ready for your review

File: spec/unit/blockscore/person_spec.rb [hunk 1]
```diff
@@ -1,10 +1,12 @@
 module BlockScore
   RSpec.describe Person do
     describe '.new' do
-      subject(:person) { BlockScore::Person.new(attributes_for(:person)) }
+      let(:attrs)      { attributes_for(:person) }
+      subject(:person) { BlockScore::Person.new(attrs) }
```
Comment (dkubb) [inline spec/unit/blockscore/person_spec.rb]: The subject is evaluated lazily, so there's no particular need to define the `:attrs` let block before the subject.

Typically I will order my spec setup like:

* subject
* let
* before
* after
* it/its blocks

The subject is the object under test, so I think it's the most important thing in the test and what should appear first.

File: spec/unit/blockscore/person_spec.rb [hunk 2]
```diff
@@ -1,10 +1,12 @@
...
 
       it { is_expected.not_to be_persisted }
       its(:class) { should be BlockScore::Person }
+      its(:question_sets) { should be_an_instance_of BlockScore::Collection }
```
Comment (dkubb) [inline spec/unit/blockscore/person_spec.rb]: I might suggest a separate `describe '#question_sets'` block with this. I generally would have the `.new` describe block focus on describing the behaviour and side effects of the specified method. The fact that there's a `#question_sets` method available on the object isn't really pertinent to the constructor.

File: spec/unit/blockscore/person_spec.rb [hunk 3]
```diff
@@ -1,10 +1,12 @@
 module BlockScore
   RSpec.describe Person do
```
Comment (dkubb) [inline spec/unit/blockscore/person_spec.rb]: As described in another PR, I would probably not monkey patch `RSpec.describe` inside the BlockScore namespace. By including it inside this namespace you're inadvertently modifying the object under test.

Also if you use the suggested `RSpec.describe BlockScore::Person`, then you can replace the places within the spec that refer to `BlockScore::Person` directly with `describe_class`.

File: spec/unit/blockscore/person_spec.rb (hunk 1 above)
Comment (tjchambers) [inline spec/unit/blockscore/person_spec.rb]: Your sequence is contrary to the convention(?) in this code, and to my experience. I always put the lets first. @backus?

File: spec/unit/blockscore/person_spec.rb (hunk 2 above)
Comment (tjchambers) [inline spec/unit/blockscore/person_spec.rb]: @dkubb it is the constructor that creates the question_sets ivar. So that was my reasoning for having it here. Is that not considered?

File: spec/unit/blockscore/person_spec.rb (hunk 3 above)
Comment (tjchambers) [inline spec/unit/blockscore/person_spec.rb]: This was suggested also by @mbj, and I tend to use `described_class` myself, but that is not the convention of the specs to date. @backus?

File: spec/unit/blockscore/person_spec.rb (hunk 1 above)
Comment (dkubb) [inline spec/unit/blockscore/person_spec.rb]: I mainly provide this suggestion because it offers a consistent guide that can be followed. I have no problem if subject comes first as long as it's consistent throughout the code. I do personally find having `subject` first makes it much easier to find when looking for the object under test -- I view `let` blocks as kind of secondary; they are always subordinate to the subject.

File: spec/unit/blockscore/person_spec.rb (hunk 2 above)
Comment (dkubb) [inline spec/unit/blockscore/person_spec.rb]: > it is the constructor that creates the question_sets ivar

I think this is a valid reason for the test to exist. Please disregard my suggestion.

File: spec/unit/blockscore/person_spec.rb (hunk 3 above)
Comment (dkubb) [inline spec/unit/blockscore/person_spec.rb]: @backus I would probably recommend we change this globally across the specs. It both removes one source of possible indeterminism (RSpec modifying a class under test so it has a different interface/behaviour as in normal usage), and allows for code simplification through things like `describe_class`.

Comment (tjchambers) [inline spec/unit/blockscore/person_spec.rb]: @backus I concur with @dkubb and would do this across the board as a separate PR. 

@dkubb is this mergeable now?
